lib/cn/hblock_builder.h

```diff
@@ -42,17 +43,25 @@ hbb_destroy(struct hblock_builder *bld);
 merr_t
 hbb_finish(
     struct hblock_builder *bld,
-    struct kvs_block *blkl,
-    uint64_t min_seqno,
-    uint64_t max_seqno,
-    uint32_t num_kblocks,
-    uint32_t num_vblocks,
-    uint32_t num_vgroups,
-    const uint8_t *hlog);
+    struct kvs_block      *blk,
+    const struct vgmap    *vgmap,
+    struct key_obj        *min_pfxp,
+    struct key_obj        *max_pfxp,
+    const uint64_t         min_seqno,
+    const uint64_t         max_seqno,
+    const uint32_t         num_kblocks,
+    const uint32_t         num_vblocks,
+    const uint32_t         num_ptombs,
+    const uint8_t         *hlog,
+    void                  *ptree,
```

Comment:
I am not following why the ptree and pgc get passed as parameters either.

Reply:
ptree is not rewritten in the hblock during a kvset split. It is duplicated from the source kvset and is passed on by the caller.

Reply:
See this comment: https://github.com/hse-project/hse/pull/325#discussion_r919119552